finder/finder.go

```diff
@@ -1,60 +1,76 @@
-package main
+package finder
 
 import (
 	"errors"
 	"fmt"
+	"log"
 	"sync"
+
+	"github.com/status-im/go-maven-resolver/fetcher"
+	"github.com/status-im/go-maven-resolver/pom"
 )
 
 type Finder struct {
 	deps         map[string]bool /* to avoid checking the same dep */
-	mtx          sync.Mutex      /* for locking access to the deps map */
-	wg           sync.WaitGroup  /* to figure out when it's done */
-	fetchers     FetcherPool     /* pool of workers for HTTP requests */
+	fetchers     fetcher.Pool    /* pool of workers for HTTP requests */
 	ignoreScopes []string        /* list of scopes to ignore */
 	recursive    bool            /* recursive resolution control */
+	l            *log.Logger     /* for logging events */
+
+	mtx sync.Mutex     /* for locking access to the deps map */
+	wg  sync.WaitGroup /* to figure out when it's done */
```

Comment:
I like how moving the `WaitGroup` to the `Finder` makes it easier to track.

Reply:
Not essential, obviously, but it is something that is makes it very clear you are dealing with a struct that needs `sync` functionality.